src/backlinks/handler.js

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {
+  internalServerError, noContent, notFound, ok,
+} from '@adobe/spacecat-shared-http-utils';
+import AhrefsAPIClient from '../support/ahrefs-client.js';
+import { retrieveSiteBySiteId } from '../utils/data-access.js';
+import { toggleWWW } from '../apex/handler.js';
+
+export default async function auditBrokenBacklinks(message, context) {
+  const { type, url: siteId, auditContext } = message;
+  const { dataAccess, log, sqs } = context;
+  const {
+    AUDIT_RESULTS_QUEUE_URL: queueUrl,
+  } = context.env;
+
+  try {
+    log.info(`Received ${type} audit request for siteId: ${siteId}`);
+
+    const site = await retrieveSiteBySiteId(dataAccess, siteId, log);
+    if (!site) {
+      return notFound('Site not found');
+    }
+
+    const auditConfig = site.getAuditConfig();
+    if (auditConfig.auditsDisabled()) {
+      log.info(`Audits disabled for site ${siteId}`);
+      return ok();
+    }
+
+    if (auditConfig.getAuditTypeConfig(type)?.disabled()) {
+      log.info(`Audit type ${type} disabled for site ${siteId}`);
+      return ok();
+    }
+
+    const ahrefsAPIClient = AhrefsAPIClient.createFrom(context);
+    const urls = [...new Set([site.getBaseURL(), toggleWWW(site.getBaseURL())])];
+
+    await Promise.all(urls.map(async (url) => {
```

Comment:
While this will run two audits, both audits will have the same audit type. This means that the second audit will overwrite the latest audit (the first one) in the latest audit table. 

You should probably store the audit result for both URLs in the auditResult of a signle added audit.

Reply:
+1 for having a single audit containing results for both www and non-www. 

Semantically speaking both domain.com and www.domain.com points to the same site instance in the star catalogue. When querying with both keys (domain.com, www.domain.com) the result should be the same

Reply:
auditResult contains now both www and non-ww results.